spectree/plugins/falcon_plugin.py

```diff
@@ -175,27 +175,33 @@ def parse_path(self, route, path_parameter_descriptions):
 
         return f'/{"/".join(subs)}', parameters
 
-    def request_validation(self, req, query, json, headers, cookies):
+    def request_validation(self, req, query, json, form, headers, cookies):
         if query:
             req.context.query = query.parse_obj(req.params)
         if headers:
             req.context.headers = headers.parse_obj(req.headers)
         if cookies:
             req.context.cookies = cookies.parse_obj(req.cookies)
-        try:
-            media = req.media
-        except self.FALCON_HTTP_ERROR as err:
-            if err.status not in self.FALCON_MEDIA_ERROR_CODE:
-                raise
-            media = None
         if json:
+            try:
+                media = req.media
+            except self.FALCON_HTTP_ERROR as err:
+                if err.status not in self.FALCON_MEDIA_ERROR_CODE:
+                    raise
+                media = None
             req.context.json = json.parse_obj(media)
+        if form:
+            # TODO - possible to pass the BodyPart here?
+            # req_form = {x.name: x for x in req.get_media()}
+            req_form = {x.name: x.stream.read() for x in req.get_media()}
+            req.context.form = form.parse_obj(req_form)
```

Comment:
@vytas7 hi, you informed me previously about multipart forms support in Falcon here in this PR. I was looking into it, but I cant make it work in a way I'd be happy with. 

Here we're parsing the request data & passing it into Pydantic model for a validation, which is then making available in a view. I'd like to pass into Pydantic model directly the items returned from `get_media()` (is it `BodyPart` right?), but I think their content is getting evaluated while passing through Pydantic model, so when trying to read them again in the view later, it's returning just an empty bytes sequence. Which is what the docstring in `get_media()` says, but I dont know how to access the cached version of the data. Any advice please?

reading in view:
https://github.com/0b01001001/spectree/blob/d6c4e744552f8af3d719bba3511f0afbd7efd659/tests/test_plugin_falcon.py#L190-L193

Reply:
Hi again @yedpodtrzitko!
Yes, Falcon's multipart support is designed for parsing the form on-the-fly, without buffering any data into the memory or temp files behind the user's back (because the former can make you run out of memory, and the latter kill performance, depending on what you're doing).

I'm not familiar with Spectree's codebase, what would you like to send into Pydantic, only the parts serialized as JSON(/YAML/msgpack/etc) and text fields, or also files? Would you like to spool file parts to tempfiles?

In general, if you want to pass a synchronous data structure into Pydantic, you'll need it to create a wrapper to store that data into when parsing.